File: JavaScriptPractice/Leetcode389.js
var findTheDifference = function(s, t) {
    let dictt={}
    let dicts={}
    for (let c of s) {
        if (dicts[c]) {
            dicts[c]++
        }
        else 
        {
            dicts[c]=1
        }
        
    }
    for (let c of t) {
        if (dictt[c]) {
            dictt[c]++
        }
        else 
        {
            dictt[c]=1
        }
    }
    for (let c of t) {
        if (!dicts[c]|| dictt[c]>dicts[c]) {
            return c;
        }
    }
    return '0'

    
};

File: CppPractice/CppLeetcode/Leetcode389.cpp
#include <map>
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<char,int> dicts;
        map<char,int> dictt;
        for(auto c : s)
        {
            dicts[c] ++;
        }
        for(auto c : t)
        {
            dictt[c] ++;
        }
        for(auto c : t)
        {
            if (dictt[c]>dicts[c])
            {
                return c;
            }
        }
        return '0';
    }
};